src/lib/commands.js

```diff
@@ -0,0 +1,274 @@
+import { TCModel, GVL } from "@iabtcf/core";
+import config from "./config";
+import {
+	encodeVendorConsentData,
+	encodeConsentData
+} from "./cookie/cookie";
+import {
+	CMP_VERSION,
+	CMP_ID,
+	COOKIE_VERSION
+} from "./init";
+import {SECTION_DETAILS, SECTION_VENDORS} from "./store";
+import log from "./log";
+
+const arrayFrom = require('core-js/library/fn/array/from');
+const COOKIE_VERSION_V1 = 1;
+
+const createCommands = (store) => {
+
+	/**
+	 * Converts vector to consent object
+	 */
+	const vectorToObject = (set) => {
+		let obj = {};
+
+		set.forEach((hasConsent, id) => {
+			obj[id] = hasConsent;
+		});
+
+		return obj;
+	};
+
+	 /**
+	 * Build consent fields object v1 from data that has already been persisted.
+	 */
+	const getConsentFieldsObject = () => {
+		const {
+			persistedConsentData
+		} = store;
+
+		const {
+			created,
+			lastUpdated,
+			cmpId_: cmpId = store.tcModel.cmpId,
+			cmpVersion_: cmpVersion = 1,
```

Comment:
I think that this value will equal `2` (because it is obtained from consent string v2) but it should be `1`. I know that correct `cmpVersion` is passed in parameter of `getConsentFieldsV1` command but what will happen if someone forget to pass this value? Also name of this function may by misleading because in fact it is used only to generate consent string v1.

Reply:
After reading the specs one more time I think in our case cmpVersion should always be 2. Because regarding to iab docs cmpVersion is the version of the CMP app on which user gave consent and transparency was establsihed. Property which is more important to us is cookieVersion and it is being sent as an argument.